src/screens/home.js

```diff
@@ -26,62 +26,76 @@ const Home = ({ session }) =>  {
   const [loading, setLoading] = useState(true)
   const [username, setUsername] = useState(null)
   const [avatar_url, setAvatarUrl] = useState(null)
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [data, setData] = useState(null); // Add this line to define 'data' variable
+
+
+  const closeModal = () => {
+  setIsModalOpen(false);
+};
 
   useEffect(() => {
     getProfile()
   }, [session])
 
-  const getProfile = async () => {
-    try {
-      setLoading(true)
-      const { user } = session
-
-      let { data, error, status } = await supabase
-        .from('profiles')
-        .select(`username, avatar_url`)
-        .eq('id', user.id)
-        .single()
-
-      if (error && status !== 406) {
-        throw error
-      }
-
-      if (data) {
-        setUsername(data.username)
-        setAvatarUrl(data.avatar_url)
-      }
-    } catch (error) {
-      alert(error.message)
-    } finally {
-      setLoading(false)
+const getProfile = async () => {
+  try {
+    setLoading(true);
+    const { user } = session;
+
+    let { data, error, status } = await supabase
+      .from('profiles')
+      .select(`username, avatar_url`)
+      .eq('id', user.id)
+      .single();
+
+    if (error && status !== 406) {
+      throw error;
+    }
+
+    if (data) {
+      setUsername(data.username);
+      setAvatarUrl(data.avatar_url);
```

Comment:
do we have functionality for users to upload their avatar?

Reply:
No not yet. Could be added tho